tests/foreman/api/test_convert2rhel.py

```diff
@@ -223,17 +217,42 @@ def oracle(
     act_key = create_activation_key(
         module_target_sat, module_entitlement_manifest_org, module_lce, cv, c2r_sub.id
     )
+    # UBI repo required for subscription-manager packages on Oracle
     ubi_url = settings.repos.convert2rhel.ubi7 if major == '7' else settings.repos.convert2rhel.ubi8
-    ubi = create_repo(module_target_sat, module_entitlement_manifest_org, ubi_url)
-    ubi_repo = ubi.full_path.replace('https', 'http')
-    register_host(
-        module_target_sat,
-        act_key,
-        module_entitlement_manifest_org,
-        smart_proxy_location,
-        oracle_host,
-        ubi_repo,
-    )
+
+    # Register Oracle host with Satellite
+    command = module_target_sat.api.RegistrationCommand(
+        organization=module_entitlement_manifest_org,
+        activation_keys=[act_key.name],
+        location=smart_proxy_location,
+        insecure=True,
+        repo=ubi_url,
+    ).create()
+    assert oracle_host.execute(command).status == 0
+
+    if oracle_host.execute('needs-restarting -r').status == 1:
+        oracle_host.power_control(state='reboot')
+
+    if major == '8':
+        # needs-restarting missing in OEL8
+        assert oracle_host.execute('dnf install -y yum-utils').status == 0
+        # Fix inhibitor CHECK_FIREWALLD_AVAILABILITY::FIREWALLD_MODULES_CLEANUP_ON_EXIT_CONFIG -
+        # Firewalld is set to cleanup modules after exit
+        result = oracle_host.execute(
+            'sed -i -- "s/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g" '
+            '/etc/firewalld/firewalld.conf && firewall-cmd --reload'
+        )
+        assert result.status == 0
+
+        # Fix inhibitor TAINTED_KMODS::TAINTED_KMODS_DETECTED - Tainted kernel modules detected
+        blacklist_cfg = '/etc/modprobe.d/blacklist.conf'
+        assert oracle_host.execute('modprobe -r nvme_tcp').status == 0
+        assert oracle_host.execute(f'echo "blacklist nvme_tcp" >> {blacklist_cfg}').status == 0
+        assert (
+            oracle_host.execute(f'echo "install nvme_tcp /bin/false" >> {blacklist_cfg}').status
+            == 0
+        )
```

Comment:
Just out of curiosity, do you know why the kernel is tainted? From the code, it looks like the `nvme_tcp` module is the issue. Why is that?

Reply:
iirc, I think it was some tech-preview feature in OEL8 as well as in EL8, but don't know much about it TBH :)